File: Programmers/PGS_64065_튜플.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    string tmp = "";
    int num[101010] = {0,};
    vector<pair<int, int>> counter;
    for(char ch : s) {
        if(ch >= '0' && ch <= '9') {
            tmp += ch;
        } else if(tmp.length()) {
            num[stoi(tmp)]++, tmp.clear();
        }
    }
    for(int i = 1; i < 101010; i++) {
        if(num[i]) {
            counter.emplace_back(num[i], i);
        }
    }
    sort(counter.begin(), counter.end());
    reverse(counter.begin(), counter.end());
    for(int i = 0; i < counter.size(); i++) {
        answer.push_back(counter[i].second);
    }
    return answer;
}

File: Programmers/PGS_64065_튜플.py
def solution(s):
    answer = []
    s = s.replace("{", "").replace("}", "")
    num = [int(n) for n in s.split(",")]
    counter = {}
    for value in num:
        if value in counter:
            counter[value] += 1
        else:
            counter[value] = 1
    counter = dict(sorted(counter.items(), key = lambda item: item[1], reverse = True))
    return list(counter.keys())